### PR TITLE
MAINT: removing outdated code branches

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description 
removing outdated code branches.
Removed branches for sklearn versions less < 1.0, since currently agreed to support sklearn >= 1.0 version
Discussed on https://github.com/intel/scikit-learn-intelex/pull/1644#discussion_r1658710364

### Justification
This reduction is required also for Array API scope estimation.

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.
- [ ] I have resolved any merge conflicts that might occur with the base branch.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Comment by icfaust** [comment]
I will not review until @napetrov or @Alexsandruss have weighed in. This is relatively similar to #1781 and therefore I would expect that a similar style analysis and discussion on the user base would be necessary before moving this PR forward. I see as well that it has been converted to draft, so this is also contingent on that as well.

**Comment by samir-nasibli** [comment]
This PR is outdated.

After discussion with Nikolai, it is agreed that, currently we are not going to remove unsupported branches, but it make sense in new file, in new development avoid adding branches for unsupported sklearn version.
@Alexsandruss @icfaust @ethanglaser just FYI
